clients/tests/gtest_main.cpp

```diff
@@ -85,6 +90,70 @@ static size_t get_system_memory_GiB()
 #endif
 }
 
+void precompile_test_kernels()
+{
+    puts("precompiling test kernels...");
+    WorkQueue<std::string> tokenQueue;
+
+    std::vector<std::string> tokens;
+    auto                     ut = testing::UnitTest::GetInstance();
+    for(int ts_index = 0; ts_index < ut->total_test_suite_count(); ++ts_index)
+    {
+        const auto ts = ut->GetTestSuite(ts_index);
+        // skip disabled suites
+        if(strncmp(ts->name(), "DISABLED", 8) == 0)
+            continue;
+        for(int ti_index = 0; ti_index < ts->total_test_count(); ++ti_index)
+        {
+            const auto  ti   = ts->GetTestInfo(ti_index);
+            std::string name = ti->name();
+            // only care about accuracy tests
+            if(name.find("vs_fftw/") != std::string::npos
+               && name.find("_batch_1_") != std::string::npos)
+            {
+                name.erase(0, 8);
+                tokens.emplace_back(std::move(name));
+            }
+        }
+    }
+
+    std::random_device dev;
+    std::mt19937       dist(dev());
+    std::shuffle(tokens.begin(), tokens.end(), dist);
```

Comment:
This is to even out the load between the processes?

Reply:
The shuffle is to try to make sure many similar test cases aren't done in parallel, which might cause us to compile the same kernel at the same time in multiple threads.  Currently the RTC code doesn't try to prevent that.  I hope to make that change eventually though, and then this shuffle could go away.